spec/operators/timeout-spec.js

```diff
@@ -1,44 +1,102 @@
-/* globals describe, it, expect */
+/* globals describe, it, expect, expectObservable, hot, rxTestScheduler */
 var Rx = require('../../dist/cjs/Rx');
 var Observable = Rx.Observable;
 
-describe('Observable.prototype.timeout', function () {
-  it('should timeout after a specified delay', function (done) {
-    Observable.never().timeout(100)
-      .subscribe(function (x) {
-        throw 'should not next';
-      }, function (err) {
-        expect(err.message).toBe('timeout');
-        done();
-      }, function () {
-        throw 'should not complete';
+describe('Observable.prototype.timeout()', function () {
+  var defaultTimeoutError = new Error('timeout');
+  
+  it('should timeout after a specified timeout period', function () {
+    var e1 = Observable.never();
+    var expected = '-----#';
+    
+    expectObservable(e1.timeout(50, null, rxTestScheduler)).toBe(expected, null, defaultTimeoutError);
+  });
+  
+  it('should timeout after specified timeout period and send the passed error', function () {
+    var e1 = Observable.never();
+    var expected = '-----#';
+    var value = 'hello';
+    
+    expectObservable(e1.timeout(50, value, rxTestScheduler)).toBe(expected, null, value);
+  });
+  
+  it('should timeout at a specified Date', function (done) {
+    var e1 = Observable.never();
+    
+    e1.timeout(new Date(Date.now() + 100), null)
```

Comment:
Absolute timeout period can cause intermittent test failure due to time frame difference in test scheduler (usually 1/10 frame difference) between expected timeout to actual timeout, by doing calculation twice (to setup fixture value create `Date` via `Date.now() + n` then actual operator does `+due - Date.now()`). 
to avoid intermittent failure, test cases covers absolute timeout are not using exact frame comparison.